src/DinosaurHydrator/DinosaurHydrator.php

```diff
@@ -22,19 +22,24 @@ public static function getAllDinos(PDO $db) : Array
         return $query->fetchAll();
     }
 
-    /** Connects the database and fetches one dinosaurs information, appending the foodType
+    /** Connects the database and fetches one dinosaur information, appending the foodType
      * table with a left join and returning a Dinosaur object
      * @param int $id
-     * @return Dinosaur
+     * @return
      */
-    public static function getDino(PDO $db, int $id) : Dinosaur
+    public static function getDino(PDO $db, int $id)
     {
         // Prepares (/stores) the criteria for data we want to retrieve from the db
-        $query = $db->prepare('SELECT `dinos`.`id`, `dinos`.`species`, `dinos`.`foodType`  AS `foodType`, `dinos`.`height`, `dinos`.`weight`, `dinos`.`length`, `dinos`.`killerRating`, `dinos`.`intelligence`, `dinos`.`age`, `dinos`.`imageUrl`, , `foodTypes`.`imageUrl` AS `logoUrl` FROM `dinos` INNER JOIN `foodTypes` ON `dinos`.`foodType` = `foodTypes`.`id` WHERE `id` = :id;');
+        $query = $db->prepare('SELECT `dinos`.`id`, `dinos`.`species`, `foodTypes`.`name`  AS `foodType`, `dinos`.`height`, `dinos`.`weight`, `dinos`.`length`, `dinos`.`killerRating`, `dinos`.`intelligence`, `dinos`.`age`, `dinos`.`imageUrl`, `foodTypes`.`imageUrl` AS `logoUrl` FROM `dinos` INNER JOIN `foodTypes` ON `dinos`.`foodType` = `foodTypes`.`id` WHERE `dinos`.`id` = :id;');
         $query->execute([':id' => $id]);
         // Sets the fetch mode (what format we get the data returned in) to the class of Dinosaur
         $query->setFetchMode(PDO::FETCH_CLASS|PDO::FETCH_PROPS_LATE, Dinosaur::class);
-        // Fetches all rows which match the criteria (as opposed to fetch() which returns one row)
-        return $query->fetch();
+        // Fetches the row which matches the criteria
+        $result = $query->fetch();
+        if ($result) {
+            return $result;
+        } else {
+            return "We couldn't find your dinosaur!";
+        }
```

Comment:
You can scrap the else here, just have the if with the return, and then the second return.

Doesn't make a big difference, but just helps simplify things slightly.

```
if ($result) {
     return $result;
} 
return "We couldn't find your dinosaur!";
        
```